### PR TITLE
[16.0][ADD] shopinvader_search_engine_update_pricelist

### PR DESCRIPTION
**Pull request by trisdoan**

### CODE REVIEW

File: shopinvader_search_engine_update_pricelist/readme/CONTRIBUTORS.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+- [Trobz](https://www.trobz.com):
```
Comment (simahawk) [inline shopinvader_search_engine_update_pricelist/readme/CONTRIBUTORS.md]: pls add me here :wink:

File: shopinvader_search_engine_update_pricelist/models/product_pricelist.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+class PricelistItem(models.Model):
+    _name = "product.pricelist.item"
+    _inherit = ["product.pricelist.item", "se.product.update.mixin"]
+
+    def get_products(self):
+        Product = self.env['product.product']
+        return Product.search(self._get_product_domain())
+    
+    def _get_product_domain(self):
```
Comment (simahawk) [inline shopinvader_search_engine_update_pricelist/models/product_pricelist.py]:
```suggestion
    def _se_get_product_domain(self):
```
and I would suggest to rename the methods of the mixin to be more explicit about their scope (eg: `get_products` seems quite generic.

**Review by simahawk — COMMENTED** [review]
LG overall

**Comment by trisdoan** [comment]
Hello @simahawk, I updated as you suggest :pray:

**Review by ivantodorovich — APPROVED** [review]

**Comment by trisdoan** [comment]
Hello @simahawk @ivantodorovich , I added a fixup to adapt _se_get_product_domain for multi record case. Please help to review :pray:

File: shopinvader_search_engine_update_pricelist/models/product_pricelist.py [hunk 3]
```diff
@@ -9,10 +9,13 @@ class PricelistItem(models.Model):
     _inherit = ["product.pricelist.item", "se.product.update.mixin"]
 
     def get_products(self):
-        Product = self.env["product.product"]
-        return Product.search(self._se_get_product_domain())
+        Products = self.env["product.product"]
+        for item in self:
+            Products |= Products.search(item._se_get_product_domain())
+        return Products
```
Comment (ivantodorovich) [inline shopinvader_search_engine_update_pricelist/models/product_pricelist.py]:
```
from odoo.osv import expression
...
domains = [rec._se_get_product_domain() for rec in self]
return Product.search(expression.OR(domains))
```

That way you perform a single SQL query

Comment (trisdoan) [reply shopinvader_search_engine_update_pricelist/models/product_pricelist.py]: Hi, thanks for your suggestion :pray:

**Review by ivantodorovich — APPROVED** [review]

**Review by simahawk — APPROVED** [review]

**Review by lmignon — APPROVED** [review]

**Pull request merged**